src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -959,6 +960,30 @@ public virtual TokenValidationResult ValidateToken(string token, TokenValidation
             if (tokenParts.Length != JwtConstants.JwsSegmentCount && tokenParts.Length != JwtConstants.JweSegmentCount)
                 return new TokenValidationResult { Exception = LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14111, token))), IsValid = false };
 
+            if (validationParameters.ConfigurationManager != null)
+            {
+                try
+                {
+                    validationParameters.Configuration = validationParameters.ConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).Result;
```

Comment:
Doing sync-over-async is meh (in this case you may also have to do an IO operation to retrieve the configuration from the remote server if it's not already cached), but mutating the `TokenValidationParameters` instance in a `ValidateToken` method - whose name doesn't indicate it has side effects - seems concerning to me, specially since this method is very likely to be called concurrently.

This problem was solved years ago by injecting the signing keys in the TVP at a higher level (in the ASP.NET Core JWT handler, the TVP are cloned before injecting the signing keys resolved from the config). Why isn't this layering no longer suitable?

Reply:
@kevinchalet we are investing in a model where we will be able to fall back to Last-Known-Good for metadata when the existing metadata is not valid.
This can happen if an identity provider releases metadata misspelling the 'issuer'.
Updating metadata happens automatically on a user configurable interval, default is 12 hours, so while the issue is being resolved, the site will be done for 1-2 hours while the problem is solved.
This model change will allow for LKG to be used for a period of time.

We should add a ConfigureAwait(false) here.

Reply:
@brentschmaltz How would we add ConfigureAwait(false) here?

I know we can do GetAwaiter().GetResult() and that would allow us to bypass getting an AggregateException, but I'm not sure if that's the intention here.

Reply:
Add the ConfigureAwait(false) like this:

` var configuration = validationParametersCopy.ConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();`

Reply:
Thanks Danny! Didn't realize you could include ConfigureAwait(false) when using a asynchronous method synchronously. 

Fixed! https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1695/commits/f5b3504e5f237e293dce12f52e01e133ace89fe3